maths/carmichael_number.py

```diff
@@ -31,15 +34,38 @@ def power(x: int, y: int, mod: int) -> int:
 
 
 def is_carmichael_number(n: int) -> bool:
```

Comment:
`n` must also be strictly positive... could we have a check to make sure that `n` is positive (and also have doctests for those cases)?